### PR TITLE
Check for ancient barriers in eth_syncing

### PR DESCRIPTION
**Pull request by deffrian**
Fixes | Closes | Resolves #

> Anything marked as optional that you didn't need to fill in, please remove it from the PR description. Choose one of the keywords above to refer to the issue this PR solves, followed by the issue number (e.g Fixes # 666). If there is no issue, remove the line. Remove this note after reading.

## Changes:
- If DownloadBodies or DownloadReceipts are enabled eth_syncing will be true until barriers is reached

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ x ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [ x ] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ x ] Yes
- [ ] No

### CODE REVIEW

**Review by LukaszRozmej — COMMENTED** [review]
Please fix the build and test scenarios with barriers at 0 as well as subscription

File: src/Nethermind/Nethermind.Facade/Eth/EthSyncingInfo.cs [hunk 1]
```diff
@@ -1,66 +1,78 @@
...
             {
-                result = SyncingResult.NotSyncing;
+                if (_syncConfig.DownloadReceiptsInFastSync &&
+                    _receiptStorage.LowestInsertedReceiptBlockNumber > _syncConfig.AncientReceiptsBarrierCalc)
+                {
+                    return new SyncingResult { IsSyncing = true };
+                }
+
+                if (_syncConfig.DownloadBodiesInFastSync &&
+                    _blockTree.LowestInsertedBodyNumber > _syncConfig.AncientBodiesBarrierCalc)
```
[the remark this answers is not in the document]
Comment (deffrian) [reply src/Nethermind/Nethermind.Facade/Eth/EthSyncingInfo.cs:64]: `AncientReceiptsBarrierCalc => Math.Max(1, Math.Min(PivotNumberParsed, Math.Max(AncientBodiesBarrier, AncientReceiptsBarrier)));`
So it least 1

[the remark this answers is not in the document]
Comment (MarekM25) [reply src/Nethermind/Nethermind.Facade/Eth/EthSyncingInfo.cs:64]: Please, test it with teku validator. You can talk with @matilote or @AntiD2ta

File: src/Nethermind/Nethermind.Facade/Eth/EthSyncingInfo.cs [hunk 2]
```diff
@@ -1,66 +1,78 @@
...
-        
+
         public bool IsSyncing()
         {
-            long bestSuggestedNumber = _blockFinder.FindBestSuggestedHeader().Number;
-            long headNumberOrZero = _blockFinder.Head?.Number ?? 0;
-            bool isSyncing = bestSuggestedNumber > headNumberOrZero + 8;
-            
-            return isSyncing;
+            return GetFullInfo().IsSyncing;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Facade/Eth/EthSyncingInfo.cs:75]: Minor: Lets make `SyncingResult` a struct instead of a class

**Review by LukaszRozmej — APPROVED** [review]
Looks good, fix pipeline, test + some minor comment

**Review by MarekM25 — APPROVED** [review]

**Pull request merged**